drivers/serial/uart_nrfx_uarte.c

```diff
@@ -283,11 +820,25 @@ static int uarte_nrfx_poll_in(struct device *dev, unsigned char *c)
  * @param dev UARTE device struct
  * @param c Character to send
  */
-static void uarte_nrfx_poll_out(struct device *dev,
-					 unsigned char c)
+static void uarte_nrfx_poll_out(struct device *dev, unsigned char c)
 {
 	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
 
+#ifdef CONFIG_UART_ASYNC_API
+	const struct uarte_nrfx_data *data = get_dev_data(dev);
+
+	if (data->async) {
+		while (data->async->tx_buf) {
+		}
+		/* Simulate tx buffer present,
+		 * this way uarte_nrfx_tx will return -EBUSY.
+		 */
+		data->async->tx_buf = &c;
```

Comment:
`uarte_nrfx_tx` will return `-EBUSY` but another `uarte_nrfx_poll_out` will just hang in the above loop. See the long comment right below for more context.